pulpcore/cli/common/generic.py

```diff
@@ -299,7 +299,10 @@ class PulpOption(click.Option):
     """
     Wrapper class for `click.Option`.
 
-    This provides a standard way of defining CLI options and helps keep consistent behavior.
+    This is the standard way of defining CLI options. In addition to `click.Option` it provides:
+    - `needs_plugin`: specify version constrains for an option
+    
+    The preferred way to use this is through the [`pulp_option`](pulpcore.cli.common.generic.pulp_option) factory.
```

Comment:
Can we keep this rather short and put the bulk of documentation with the `pulp_option`? Maybe "Pulp-CLI specific subclass of click.Option" and then the reference to pulp_option.

Also can you figure out what else besides `needs_plugin` this class provides?

Reply:
Sure, will do.

Hmm, it seems to provide basic templating to the help msg, so it can access the entity name that it belongs to with curly braces.

Reply:
Yes! It does. And there's another mechanism that allows to limit the use of some options to certain entity contexts. That one is probably the most complicated to explain.

Reply:
I've added an example in the docstring to provide more straighfoward way of understanding `PulpOption` features. Could you help adding one about that limiting-use-of-context feature?